packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -74,6 +148,12 @@ export const useSignUpContext = (): SignUpContextType => {
     [],
   );
 
+  if (!context || context.componentName !== 'SignUp') {
```

Comment:
For simplicity we retain the componentName check. In a future PR we can refactor this out since we always know that we're accessing a value from the expected Context.